diff --git a/dist/internals/wrapConnectorHooks.js b/dist/internals/wrapConnectorHooks.js
index 5fa7fe8d6c1a280b66b26766b227cc8994f65b7b..a28f930b042e1e77a9e51d5299a4718e1e6d3be5 100644
--- a/dist/internals/wrapConnectorHooks.js
+++ b/dist/internals/wrapConnectorHooks.js
@@ -53,7 +53,7 @@ function setRef(ref, node) {
     }
 }
 function cloneWithRef(element, newRef) {
-    const previousRef = element.ref;
+    const previousRef = element.props?.ref ?? element.ref;
     invariant(typeof previousRef !== 'string', 'Cannot connect React DnD to an element with an existing string ref. ' + 'Please convert it to use a callback ref instead, or wrap it into a <span> or <div>. ' + 'Read more: https://reactjs.org/docs/refs-and-the-dom.html#callback-refs');
     if (!previousRef) {
         // When there is no ref on the element, use the new ref directly
